clang/lib/Sema/SemaBounds.cpp

```diff
@@ -675,22 +714,27 @@ namespace {
       }
   };
 
-  // If an original value is provided, ReplaceVariableReferences returns
-  // an expression that replaces all uses of the variable V in E with the
-  // original value.  If no original value is provided and E uses V,
-  // ReplaceVariableReferences returns nullptr.
-  Expr *ReplaceVariableReferences(Sema &SemaRef, Expr *E, DeclRefExpr *V,
-                                  Expr *OriginalValue,
-                                  CheckedScopeSpecifier CSS) {
-    // Don't transform e if it does not use the value of v.
-    if (!VariableOccurrenceCount(SemaRef, V, E))
+  // If an original value is provided, ReplaceLValue returns an expression
+  // that replaces all uses of the lvalue expression LValue in E with the
+  // original value.  If no original value is provided and E uses LValue,
+  // ReplaceLValue returns nullptr.
+  Expr *ReplaceLValue(Sema &SemaRef, Expr *E, Expr *LValue,
+                      Expr *OriginalValue,
+                      CheckedScopeSpecifier CSS) {
+    // Don't transform E if does not use the value of LValue.
+    if (!FindLValue(SemaRef, LValue, E))
       return E;
 
+    // If E uses the value of LValue, but no original value is provided,
+    // we know the result is null without needing to transform E.
+    if (!OriginalValue)
```

Comment:
Can the check for `OriginalValue` move before the `FindLValue` check because this check is much cheaper than `FindLValue`?

Reply:
We need to know whether `E` uses the value of `LValue` regardless of whether `OriginalValue` is null. If `OriginalValue` is null and `FindLValue` returns true, we can return null without transforming `E`. If `OriginalValue` is non-null, we still need to call `FindLValue` so we can determine whether we can just return `E` without calling `ReplaceLValue`.